lib/core/src/Cardano/Wallet/Api/Server.hs

```diff
@@ -911,12 +913,12 @@ postSharedWalletFromRootXPrv ctx generateKey body = do
     pwd = preparePassphrase EncryptWithPBKDF2 $ getApiT (body ^. #passphrase)
     rootXPrv = generateKey (seed, secondFactor) pwd
     g = defaultAddressPoolGap
-    accIx = Index $ getDerivationIndex $ getApiT (body ^. #accountIndex)
+    ix = getApiT (body ^. #accountIndex)
     wid = WalletId $ digest $ publicKey rootXPrv
     pTemplate = scriptTemplateFromSelf (getRawKey accXPub) $ body ^. #paymentScriptTemplate
     dTemplateM = scriptTemplateFromSelf (getRawKey accXPub) <$> body ^. #delegationScriptTemplate
     wName = getApiT (body ^. #name)
-    accXPub = publicKey $ deriveAccountPrivateKey pwd rootXPrv accIx
+    accXPub = publicKey $ deriveAccountPrivateKey pwd rootXPrv (Index $ getDerivationIndex ix)
```

Comment:
Why this change?

Reply:
this is just code reshuffling allowing to use guardHardened. No logic was changed